YvesIraguha/lost-and-found-api
# #172261448 Set-up Docker

#### What does this PR do?
- Set up docker & docker-compose
#### Description of Task to be completed?
- create docker file
- create a docker image
- refactor tests
- create docker-compose file
- test containerization technology
#### How should this be manually tested?
- Clone the repository, and `cd` into the root folder of the repository
- Checkout the branch `ch-setup-docker-#172261448`
- Install docker in your PC from [docker installation](https://docs.docker.com/get-docker/)
- After installing docker, open the command line in root folder of the repository and run the command `docker-compose up`, you should see logs showing the progress on the creation of the images and setting up containers. 
- After the process is done, open your browser and browse to `http://localhost:8080/api/v1/` you should see a response with a message in the body `Welcome to lost and found API. Thanks`
- Open your client application for instance postman and send a post request to `http://localhost:8080/api/v1/lost` with ```{name:"ID Card", description:"It is an id card of someone who lost it in this road and that road"}``` You should see a response which looks like ```{message:value, item:{_id:value, name:value, description:value}}```

#### Any background context you want to provide?
This chore is done so that we can have consistency in the development process. 
#### What are the relevant pivotal tracker stories?
#172261448
